### PR TITLE
[node:buffer]: Add the `File` class

### PR DESCRIPTION
**Pull request by vansergen**
## Changes:

- Add the [`File`](https://nodejs.org/dist/latest-v18.x/docs/api/buffer.html#class-file ) class.

### Checklist

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node/buffer.d.ts [hunk 1]
```diff
@@ -163,6 +163,35 @@ declare module 'buffer' {
...
+        endings?: 'native' | 'transparent';
+        /** The File content-type. */
+        type?: string;
+        /** The last modified date of the file. `Default`: Date.now(). */
+        lastModified?: number;
+    }
+    /**
+     * A [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) provides information about files.
+     * @experimental
+     * @since v18.13.0, v19.2.0
```
Comment (peterblazejewicz) [inline types/node/buffer.d.ts:180]:
```suggestion
     * @since v18.13.0
```

File: types/node/buffer.d.ts [hunk 2]
```diff
@@ -163,6 +163,35 @@ declare module 'buffer' {
...
+     */
+    export class File extends Blob {
+        constructor(sources: Array<BinaryLike | Blob>, fileName: string, options?: FileOptions);
+        /**
+         * The name of the `File`.
+         * @since v18.13.0, v19.2.0
```
Comment (peterblazejewicz) [inline types/node/buffer.d.ts:186]:
```suggestion
         * @since v18.13.0
```

File: types/node/buffer.d.ts [hunk 3]
```diff
@@ -163,6 +163,35 @@ declare module 'buffer' {
...
+         */
+        name: string;
+        /**
+         * The last modified date of the `File`.
+         * @since v18.13.0, v19.2.0
```
Comment (peterblazejewicz) [inline types/node/buffer.d.ts:191]:
```suggestion
         * @since v18.13.0
```
we're not yet there on DT

File: types/node/buffer.d.ts [hunk 4]
```diff
@@ -163,6 +163,35 @@ declare module 'buffer' {
...
+     * @experimental
+     * @since v18.13.0, v19.2.0
+     */
+    export class File extends Blob {
+        constructor(sources: Array<BinaryLike | Blob>, fileName: string, options?: FileOptions);
+        /**
+         * The name of the `File`.
+         * @since v18.13.0, v19.2.0
+         */
+        name: string;
```
Comment (peterblazejewicz) [inline types/node/buffer.d.ts:188]:
```suggestion
        readonly name: string;
```
should this one be `readonly` as for other fields, e.g. `Blob.size?

File: types/node/buffer.d.ts [hunk 5]
```diff
@@ -163,6 +163,35 @@ declare module 'buffer' {
...
+        /**
+         * The last modified date of the `File`.
+         * @since v18.13.0, v19.2.0
+         */
+        lastModified: number;
```
Comment (peterblazejewicz) [inline types/node/buffer.d.ts:193]:
```suggestion
        readonly lastModified: number;
```
_ditto_ about readonly

File: types/node/buffer.d.ts (hunk 4 above)
Comment (vansergen) [reply types/node/buffer.d.ts]: That is correct. Node.js docs is missing this part

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@vansergen thanks!

**Comment by vansergen** [comment]
Ready to merge

**Pull request merged**